backend/src/Controllers/PostsImportController.php

```diff
@@ -0,0 +1,337 @@
+<?php
+
+namespace FL\Assistant\Controllers;
+
+use FL\Assistant\System\Contracts\ControllerAbstract;
+use FL\Assistant\System\View;
+use WP_REST_Server;
+
+/**
+ * REST API logic for importing posts.
+ */
+class PostsImportController extends ControllerAbstract {
+
+	protected $view;
+
+	public function __construct( View $view ) {
+		$this->view = $view;
+	}
+
+	/**
+	 * Register routes.
+	 */
+	public function register_routes() {
+		$this->route(
+			'/posts/import',
+			[
+				[
+					'methods'             => WP_REST_Server::CREATABLE,
+					'callback'            => [ $this, 'import_posts' ],
+					'permission_callback' => function () {
+						return current_user_can( 'edit_published_posts' );
+					},
+				],
+			]
+		);
+	}
+
+	/*Import posts */
+
+	public function import_posts( $request ) {
+		$file = $request->get_file_params();
+
+		global $wpdb;
+
+		//$uri = get_template_directory_uri().'/xmlupload/';
+		$uri         = wp_upload_dir();
+		$target_dir  = $uri;
+		$target_file = $target_dir['path'] . basename( $file['file']['name'] );
+		$filename    = basename( $file['file']['name'] );
+		$filetypenew = wp_check_filetype( $filename );
+		$upload_ok    = 1;
+		$file_type    = pathinfo( $target_file, PATHINFO_EXTENSION );
+		//$FileType = pathinfo($filetype,PATHINFO_EXTENSION);
+
+		// Check file size
+
+		// Allow certain file formats
+		if ( $file_type != 'xml' ) {
+
+			return rest_ensure_response(
+				[
+					'error'   => true,
+					'message' => 'Sorry, only XML files are allowed.',
```

Comment:
Can you use WordPress translation functions (e.g `__` or `sprintf`) anywhere there is human readable text? We need to be sure to always do that on the front or backend. There are a number of places that need it in this file.